pkg/processor/processor.go

```diff
@@ -125,7 +125,10 @@ func (p *Processor) refreshClusters() error {
 				Scheduler: snapshot.New(
 					cluster.Spec.Snapshot.BucketName,
 					cluster.Spec.Snapshot.CronSchedule,
-					cluster.Spec.Snapshot.SchedulerEnabled),
+					cluster.Spec.Snapshot.SchedulerEnabled,
+					cluster.Spec.Snapshot.Authentication.UserName,
```

Comment:
If using an image that is secured with x-pack/searchguard the operator will not be able to create snapshots. This will allow the operator to authenticate to the endpoint using basic auth.